_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_vaapi.cpp

```diff
@@ -1324,6 +1324,18 @@ mfxStatus VAAPIEncoder::QueryStatus(
 
     VASurfaceStatus surfSts = VASurfaceSkipped;
 
+#if VA_CHECK_VERSION(1,9,0)
+    vaSts = vaSyncBuffer(m_vaDisplay, codedBuffer, VA_TIMEOUT_INFINITE);
+
+    vaSts = vaQuerySurfaceStatus(m_vaDisplay, waitSurface, &surfSts);
```

Comment:
Is this call attempt to resolve https://github.com/Intel-Media-SDK/MediaSDK/issues/2392? If yes, then this is incorrect fix for the reason that `vaQuerySurfaceStatus` is an optional call. Effectively, if vaSyncBuffer (or surface) returned OK, then user can proceed and everything should work. `vaQuerySurfaceStatus` is kind of alternative way to sync in a polling fashion.

If this code really fix the issue then you make some machine state update in `vaQuerySurfaceStatus` which is missed in `vaSyncBuffer`  - and this is an actual bug.

Reply:
Yes, this resolves #2392

Reply:
Ok. As I noted, this looks to be a wrong fix. It merely highlights an issue in the driver than properly resolves it. Which operations are hidden inside vaQuerySurfaceStatus? why they were not executed within vaSyncBuffer?

Reply:
There are the same difference as between syncSurface and syncBuffer - waiting for pCurrentFrameSemaphore

Reply:
It doesn't make sense from MSDK perspective. Here MSDK VP9e code waits for output resource - bitstream.

Reply:
syncSurface actually do DdiMedia_StatusCheck first, then the same as syncBuffer